File: selection-sort.c
#include <stdio.h>

void SelectionSort(int *A, int n) {
	int i, j;
	int min, index;
	int tmp;

	for (i = 0; i < n - 1; i++) {
		min = A[i];
		index = i;

		for (j = i + 1; j < n; j++)
			if (A[j] < min) {
				min = A[j];
				index = j;
			}
		tmp = A[i];
		A[i] = A[index];
		A[index] = tmp;
	}
}

int main() {
	int A[6] = {6, 5, 5, 3, 100, 1};
	
	SelectionSort(A, 6);
	printf("%d, %d, %d, %d, %d, %d\n", A[0], A[1], A[2], A[3], A[4], A[5]);
	return 0;
}

File: selection_sort.py
#!/usr/bin/env python
# encoding: utf-8


def selection_sort(array: list):
    """
    Inplace sort
    Consider sorting n numbers stored in array A by first finding the smallest element of A
    and exchanging it with the element in A[0] . Then find the second smallest element of A, and exchange it with A[1].
    Continue in this manner for the first n - 1 elements of A.
    :param array:
    :return:
    """
    n = len(array)
    for i in range(n - 1):
        minimum = array[i]
        index = i
        for j in range(i + 1, n):
            if minimum > array[j]:
                minimum = array[j]
                index = j
        array[i], array[index] = array[index], array[i]
